server/src/main/java/org/elasticsearch/script/DeprecationMap.java

```diff
@@ -64,7 +67,7 @@ public boolean containsValue(final Object value) {
     public Object get(final Object key) {
         String deprecationMessage = deprecations.get(key);
         if (deprecationMessage != null) {
-            deprecationLogger.deprecated(deprecationMessage);
+            deprecationLogger.deprecatedAndMaybeLog(logKeyPrefix + "_" + key, deprecationMessage);
```

Comment:
nit: i would use `-` as the separator, so there is a clear distinction with the map key, since all our context names use underscores.

Reply:
Done.